### DISCUSSION TITLE
When was the co-modeling feature added to GAMA?

### DISCUSSION THREAD

**Discussion by Aya-Badawy89** [gama-platform/gama#94] [Q&A] [answered]
Hello GAMA team,

I came across this paper https://www.jasss.org/23/1/4.html where researchers are developing an extension for coupling models in NetLogo. In their literature review, they criticized that GAMA does not allow model coupling!
_"This allows GAMA to address cross-level interaction and, to some degree, dynamic adaptation of level of detail, but not heterogeneous model coupling._"
Check paragraph 2.2

I wanted to email the author to kindly point out that GAMA indeed has coupling feature, but I want to make sure that the feature was added to GAMA before this paper. 
This paper was submitted in 2018. I can find a paper by Huynh, @AlexisDrogoul , and Cambier since 2014, but I am not sure when the feature was really added to GAMA.

Thanks in advance

**Comment by hqnghi88** [comment, marked as answer]
Hi, i did not read all the paper, but if they said that , it should base on some criteria, because a feature is not always the same meaning.
For comodeling in gama, the paper you found is the first time we (i) publish to internal community. The feature was started from 2012 and worked well at 2014. Then the official announce may be in gama version 1.7 (or 1.7.1), of course it was after my thesis defend (Dec 2016).

**Reply by Aya-Badawy89** [reply]
They did not use the term feature. I used it here to explain. 
Thank you very much for your reply.
